src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -419,6 +419,72 @@ export class WorkspacesUsecases {
     await this.workspaceRepository.deleteInviteBy({ id: invite.id });
   }
 
+  async getUserInvites(user: User, limit: number, offset: number) {
+    const invites = await this.workspaceRepository.findInvitesBy(
+      {
+        invitedUser: user.uuid,
+      },
+      limit,
+      offset,
+    );
+
+    const invitesWithWorkspaceData = await Promise.all(
+      invites.map(async (invite) => {
+        const workspace = await this.workspaceRepository.findById(
+          invite.workspaceId,
+        );
+
+        return {
+          ...invite,
+          workspace: workspace.toJSON(),
+        };
+      }),
+    );
+
+    return invitesWithWorkspaceData;
+  }
+
+  async getWorkspacePendingInvitations(
+    workspaceId: WorkspaceAttributes['id'],
+    limit: number,
+    offset: number,
+  ) {
+    const invites = await this.workspaceRepository.findInvitesBy(
+      {
+        workspaceId,
+      },
+      limit,
+      offset,
+    );
+
+    const sharedWithUuids = invites.map((invite) => invite.invitedUser);
+
+    const [users, preCreatedUsers] = await Promise.all([
+      this.userUsecases.findByUuids(sharedWithUuids),
+      this.userUsecases.findPreCreatedUsersByUuids(sharedWithUuids),
+    ]);
+
+    const usersWithAvatars = await Promise.all(
+      users.map(async (user) => {
+        const avatar = user.avatar
+          ? await this.userUsecases.getAvatarUrl(user.avatar)
+          : null;
+        return {
+          ...user,
+          avatar,
+        };
+      }),
+    );
+
+    return invites.map((invite) => ({
+      ...invite,
+      user: usersWithAvatars.find((user) => invite.invitedUser === user.uuid),
+      preCreatedUser: preCreatedUsers
+        .find((user) => invite.invitedUser === user.uuid)
+        ?.toJSON(),
```

Comment:
What is this precreatedUser field returned for? 
Will we need to differentiate a user from a precreated user to display and manage invitations?

Reply:
Hey, I merged both precreated users and users so they use the same name. However, the difference between these two is that pre created users are external users  that do not have internxt accounts. I added a boolean field so you can differentiate between external and internal invitations.